infra/main.bicep

```diff
@@ -486,6 +486,17 @@ module openAiRoleBackend 'core/security/role.bicep' = if (openAiHost == 'azure')
   }
 }
 
+module openAiRoleSearchservice 'core/security/role.bicep' = if (openAiHost == 'azure') {
```

Comment:
I think we should only create these roles if necessary - i.e. only if they've enabled the vectorization feature.